src/utils.ts

```diff
@@ -1,8 +1,25 @@
 import { lstatSync, accessSync, constants, readFileSync } from "fs";
 import { createHash } from "crypto";
+import { tmpdir } from "os";
 import { join } from "pathe";
 import type { PackageJson } from "pkg-types";
 
+// Workaround for pnpm setting an incorrect `TMPDIR`.
+// Set `JITI_TMPDIR_KEEP` to a truthy value to disable this workaround.
+// https://github.com/pnpm/pnpm/issues/6140
+// https://github.com/unjs/jiti/issues/120
+export function getCacheDir() {
+  if (process.env.JITI_TMPDIR_KEEP) {
```

Comment:
We could support env always as `JITI_CACHE_DIR` :D

Reply:
I think that could be a separate feature to this workaround, because replaceing `JITI_TMPDIR_KEEP` with `JITI_CACHE_DIR` would require users to give their cache dir twice (once `TMPDIR` and once `JITI_CACHE_DIR`) if they'd want to work around the pnpm issue. I'd favor providing the workaround by default and allowing to fallback to a custom `TMPDIR` with a toggle.
Again, I understand that a `JITI_CACHE_DIR` sounds good, but can be implemented separately.